Exercises/Excercise22/solution22.java

```diff
@@ -0,0 +1,106 @@
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+class Student{
+    String student_name;
+    Integer student_mark;
+    public Student(String s,Integer r){
+        this.student_name=s;
+        this.student_mark=r;
+
+    }
+    public void setStudent_mark(Integer student_mark) {
+        this.student_mark = student_mark;
+    }
+
+    public void setStudent_name(String student_name) {
+        this.student_name = student_name;
+    }
+
+    public Integer getStudent_mark() {
+        return student_mark;
+    }
+
+    public String getStudent_name() {
+        return student_name;
+    }
+
+    @Override
+    public String toString() {
+        return "Student{" +
+                "student_name='" + student_name + '\'' +
+                ", student_mark=" + student_mark +
+                '}';
+    }
+}
+class JavaStreamAdvanceWorker{
+    public boolean odd_Or_Even(Integer o) {
+        if(o%2==1)
+            return true;
+        else
+            return false;
+    }
+    public Integer cube(Integer n){
+
+        return n*n*n;
+    }
+    public List<String> passMark(List<Student> d,Integer passMarkThreshold){
+
+        List<String> res = d.stream().filter(v ->v.student_mark>passMarkThreshold).map(v -> v.getStudent_name()).collect(Collectors.toList());
+        return res;
+
+    }
+    public Optional<Student> firstRankStudent(List<Student> r){
+        Optional<Student> fr = r.stream().max((o1, o2) -> o1.student_mark.compareTo(o2.student_mark));
+        return fr;
+    }
+    public Integer numberOfFailures(List<Student> d,Integer passMarkThreshold){
+
+        Integer res = Math.toIntExact(d.stream().filter(s -> s.student_mark < passMarkThreshold).count());
+        return res;
+
+    }
+    public void printlastFiveStudents(List<Student> s){
+        int count= (int) s.stream().count();
+        List<Student> t = s.stream().skip(count-5).collect(Collectors.toList());
```

Comment:
Use the same stream line to print

Reply:
Also write code to handle the case for the number of elements less than five use if else

Reply:
Stream print https://www.geeksforgeeks.org/how-to-print-elements-of-a-stream-in-java-8/